### PR TITLE
Add notes about relations between processes

### PR DESCRIPTION
**Pull request by amontoison**
https://juliasmoothoptimizers.github.io/Krylov.jl/previews/PR685/processes/

### CODE REVIEW

File: docs/src/processes.md [hunk 1]
```diff
@@ -184,6 +224,10 @@ The function [`golub_kahan`](@ref golub_kahan) returns $V_{k+1}$, $U_{k+1}$ and
 
 Related methods: [`LNLQ`](@ref lnlq), [`CRAIG`](@ref craig), [`CRAIGMR`](@ref craigmr), [`LSLQ`](@ref lslq), [`LSQR`](@ref lsqr) and [`LSMR`](@ref lsmr).
 
+!!! note
+    The Golub-Kahan process coincides with the Hermitian Lanczos process applied on the normal equations $A^HA = A^Hb$ and $AA^H = b$.
```
Comment (dpo) [inline docs/src/processes.md:228]:
```suggestion
    The Golub-Kahan process coincides with the Hermitian Lanczos process applied to the normal equations $A^HA x = A^Hb$ and $AA^H x = b$.
```

File: docs/src/processes.md [hunk 2]
```diff
@@ -184,6 +224,10 @@ The function [`golub_kahan`](@ref golub_kahan) returns $V_{k+1}$, $U_{k+1}$ and
...
+    It is also equivalent to the Hermitian Lanczos process applied on $\begin{bmatrix} 0 & A \\ A^H & 0 \end{bmatrix}$ with the initial vector $\begin{bmatrix} b \\ 0 \end{bmatrix}$.
```
Comment (dpo) [inline docs/src/processes.md:229]:
```suggestion
    It is also related to the Hermitian Lanczos process applied to $\begin{bmatrix} 0 & A \\ A^H & 0 \end{bmatrix}$ with initial vector $\begin{bmatrix} b \\ 0 \end{bmatrix}$.
```

File: docs/src/processes.md [hunk 3]
```diff
@@ -230,6 +274,9 @@ Related methods: [`USYMLQ`](@ref usymlq), [`USYMQR`](@ref usymqr), [`TriLQR`](@r
 saunders_simon_yip
 ```
 
+!!! note
+    The Saunders-Simon-Yip is equivalent to the block-Lanczos process applied on $\begin{bmatrix} 0 & A \\ A^H & 0 \end{bmatrix}$ with the initial matrix $\begin{bmatrix} b & 0 \\ 0 & c \end{bmatrix}$.
```
Comment (dpo) [inline docs/src/processes.md:278]:
```suggestion
    The Saunders-Simon-Yip is equivalent to the block-Lanczos process applied to $\begin{bmatrix} 0 & A \\ A^H & 0 \end{bmatrix}$ with initial matrix $\begin{bmatrix} b & 0 \\ 0 & c \end{bmatrix}$.
```

File: docs/src/processes.md [hunk 4]
```diff
@@ -278,6 +325,10 @@ The function [`montoison_orban`](@ref montoison_orban) returns $V_{k+1}$, $H_{k+
 
 Related methods: [`GPMR`](@ref gpmr).
 
+!!! note
+    The Montoison-Orban is equivalent to the block-Arnoldi process applied on $\begin{bmatrix} 0 & A \\ B & 0 \end{bmatrix}$ with the initial matrix $\begin{bmatrix} b & 0 \\ 0 & c \end{bmatrix}$.
```
Comment (dpo) [inline docs/src/processes.md:329]:
```suggestion
    The Montoison-Orban is equivalent to the block-Arnoldi process applied to $\begin{bmatrix} 0 & A \\ B & 0 \end{bmatrix}$ with initial matrix $\begin{bmatrix} b & 0 \\ 0 & c \end{bmatrix}$.
```

**Comment by amontoison** [comment]
Thanks for the review @dpo!

**Pull request merged**